File: leetcode.com/3386_button_with_longest_push_time/solution.c
int buttonWithLongestTime(int **events, int eventsSize, int *eventsColSize)
{
    int best_index = events[0][0];
    int best_duration = events[0][1];

    int prev = events[0][1];

    for (int i = 1 ; i < eventsSize ; i++) {
        int duration = events[i][1] - prev;

        if (duration == best_duration && events[i][0] < best_index) {
            best_index = events[i][0];
        }

        if (duration > best_duration) {
            best_index = events[i][0];
            best_duration = duration;
        }

        prev = events[i][1];
    }

    return best_index;
}

File: leetcode.com/3386_button_with_longest_push_time/solution.py
class Solution:
    def buttonWithLongestTime(self, events: List[List[int]]) -> int:
        longest_idx = events[0][0]
        longest_time = events[0][1]

        prev = events[0][1]
        for idx, time in events[1:]:
            d = time - prev

            if d == longest_time:
                longest_idx = min(longest_idx, idx)

            if d > longest_time:
                longest_time = d
                longest_idx = idx

            prev = time

        return longest_idx
